lib/Inspection/StatusT.h

```diff
@@ -0,0 +1,95 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <type_traits>
+
+#include <optional>
+
+#include "Inspection/Status.h"
+
+namespace arangodb::inspection {
+
+template<typename T>
+struct StatusT {
```

Comment:
`T` could be requested not to be derived/convertible from/to `Status`.

Reply:
What is the advantage of `StatusT<Status>` compared with `Status`?

Reply:
I think `StatusT<Status>` would only cause confusion and should therefore not be allowed.

Reply:
I wouldn't disallow it. Yes, it's confusing, but it also might come up naturally when composing interfaces, and I don't see why someone would do it if not necessary (neither by accident nor on purpose), so I think there's no harm in keeping the option.

Reply:
`StatusT<Status>` just means that you have an operation that *either* yields an error, which is reflected in the *error* part of the variant, and happens to be a `Status`, *or* a successful operation that is reflected in the *get* part of the variant which happens to be a `Status`.

Sure, it might be a bad life choice to do this, but it unfortunately comes up sometimes. I don't want to stand in programmer's way too much (in particular in view of `ErrorT` introduced in #17386)